helix-loader/src/lib.rs

```diff
@@ -26,6 +28,11 @@ pub fn initialize_config_file(specified_file: Option<PathBuf>) {
     CONFIG_FILE.set(config_file).ok();
 }
 
+pub fn initialize_log_file(specified_file: PathBuf) {
```

Comment:
Like `initialize_config_file`, this can take an `Option<PathBuf>` and we can move the directory creation (from [here](https://github.com/helix-editor/helix/blob/f6d65cf089a0b845ef01b91d0424a9ffe1e9ea45/helix-term/src/main.rs#L45-L48)) into that block. That way we don't create the cache_dir unless we use it